include/nbl/asset/utils/IShaderCompiler.h

```diff
@@ -354,20 +354,28 @@ class NBL_API2 IShaderCompiler : public core::IReferenceCounted
 						lookupHash = std::hash<core::blake3_hash_t>{}(hash);
 					}
 
+					// Making an entry to insert into write cache
+					inline SEntry(const SEntry& other, dependency_container_t dependencies, core::smart_refctd_ptr<asset::ICPUBuffer> spirv,
+						core::blake3_hash_t uncompressedContentHash, size_t uncompressedSize)
+						: mainFileContents(other.mainFileContents), compilerArgs(other.compilerArgs), hash(other.hash),
+						lookupHash(other.lookupHash), dependencies(dependencies), spirv(spirv),
+						uncompressedContentHash(uncompressedContentHash), uncompressedSize(uncompressedSize) {}
```

Comment:
but this  kinda does nothing?

its still inline and merely sets the members to values you pass in.

My understanding would be that you no longer make it inline, and MOVE the LZMA compression INTO this constructor

Also if `other` can be overwritten then its better to have a non-const method like `setContent(dependency_container_t&& dependencies, const ICPUShader* spirv)` that will fill out all the stuff